giraffe/src/components/SizedPlot.tsx

```diff
@@ -91,18 +83,38 @@ export const SizedPlot: FunctionComponent<SizedPlotProps> = ({
 
   const defaultSpec = env.getSpec(0)
 
-  const valueX = env.xScale.invert(hoverEvent.x)
-  let clampedValueX = NaN
+  // get me the x value (which might be a timestamp) of a particular
+  // point on the x axis
+  const getValueX = (point, snapToDataPoint = true) => {
```

Comment:
boolean arguments are usually a smell, and I think this is a case of premature abstraction. To make this function work, we basically have to call it in three different configurations in the three places its called. In my opinion, it just obfuscates the meaning of code and creates a needless function, since we're basically replicating the three lines the function replaced.

We could get the same meaning and the same relationship in the same number of lines, without the need of an extra function to refer to, by just having a couple of first class function calls.

The function isn't needed in this call at all, since the value returned from `env.xScale.invert` is used:
```ts
const valueX = getValueX(hoverEvent.x, false)
const clampedValueX = getNearestTimeStamp(valueX)
```
could then become:
```ts
const valueX = env.xScale.invert(hoverEvent.x)
const clampedValueX = getNearestTimeStamp(valueX)
```
It's easier to read - you don't have to go to a function and see that `env.xScale.invert` is called, you just see it called here.

---

This code
```ts
const beginning = getValueX(xRange[0], true)
const end = getValueX(xRange[1], true)
```
could be changed to something like this:

```ts
const beginning =  getNearestTimeStamp(env.xScale.invert(xRange[0]))
const end = getNearestTimeStamp(getValueX(xRange[1]))
```

This code is the same number of lines, and communicates the same thing, but doesn't require a second function wrapped around `invert`.

I think this function is one of those cases where being prematurely DRY actually makes the code harder to comprehend.

Reply:
whereas the 'invert' function is what is confusing to me; hence why I renamed it.  and you did include 'getValueX' in the second line of what you suggested (assume that was a mistake....)

Reply:
i understand wanting less lines of code, but the 'invert' makes no sense to me; it isn't explained/documented anywhere.
......pondering.....

Reply:
i propose that getNearestTimeStamp be renamed to getNearestDataPoint
which is more accurate
and it takes the actual graph point and does the env.xScale.invert

because it is a smell for the method to have submethods called within its arguments.
that way, it will be clearer what is going on,
and the 'getValueX' wrapper will be removed

Reply:
fair enough! thanks for listening